src/effects/underwater.c

```diff
@@ -1,14 +1,135 @@
 #include "common.h"
 #include "effects_internal.h"
 
+void underwater_init(EffectInstance* effect);
+void underwater_update(EffectInstance* effect);
+void underwater_render(EffectInstance* effect);
 void underwater_appendGfx(void* effect);
 
-INCLUDE_ASM(s32, "effects/underwater", underwater_main);
+EffectInstance* underwater_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, s32 arg5) {
+    EffectBlueprint bp;
+    EffectInstance* effect;
+    UnderwaterFXData* data;
+    s32 numParts = 1;
+    s32 i;
+    s32 j;
 
-void underwater_init(void) {
+    bp.init = underwater_init;
+    bp.update = underwater_update;
+    bp.renderWorld = underwater_render;
+    bp.unk_00 = 0;
+    bp.unk_14 = NULL;
+    bp.effectID = EFFECT_UNDERWATER;
+
+    effect = shim_create_effect_instance(&bp);
+    effect->numParts = numParts;
+    data = effect->data.underwater = shim_general_heap_malloc(numParts * sizeof(*data));
+    ASSERT(effect->data.underwater != NULL);
+
+    data->unk_00 = arg0;
+    data->unk_14 = 0;
+    if (arg5 <= 0) {
+        data->unk_10 = 1000;
+    } else {
+        data->unk_10 = arg5;
+    }
+    data->unk_1F = 0;
+    data->unk_04 = arg1;
+    data->unk_08 = arg2;
+    data->unk_0C = arg3;
+    data->unk_18 = arg4;
+    data->unk_1C = 10;
+    data->unk_1D = 110;
+    data->unk_1E = 255;
+    data->unk_20 = 190;
+    data->unk_21 = 220;
+    data->unk_22 = 255;
+
+    for (i = 0; i < 19; i++) {
```

Comment:
ARRAY_COUNT for theez pleez

Reply:
Wasn't sure if I should use ARRAY_COUNT on these because the count applies for two different arrays. Should I just pick `unk_23` then?